packages/kbn-optimizer/src/report_optimizer_stats.ts

```diff
@@ -35,7 +39,23 @@ export function reportOptimizerStats(reporter: CiStatsReporter, name: string) {
         }
 
         if (n.kind === 'C' && lastState) {
-          await reporter.metric('@kbn/optimizer build time', name, lastState.durSec);
+          await reporter.metrics([
+            {
+              name: '@kbn/optimizer build time',
+              subName: name,
+              value: lastState.durSec,
+            },
+            ...config.bundles.map(bundle => {
+              // make the cache read from the cache file since it was likely updated by the worker
+              bundle.cache.refresh();
+
+              return {
+                name: `@kbn/optimizer bundle module count`,
+                subName: bundle.id,
+                value: bundle.cache.getModuleCount() || 0,
```

Comment:
I think the module count is a useful metric to understand how imports are impacting the build, it's not uncommon for an innocent looking import to trigger the addition of dozens of modules, and I think it would be good for us to keep an eye on that personally. We also plan to keep an eye on the page overhead and the size of assets that get loaded in the browser regularly in https://github.com/elastic/kibana/issues/65524